include/ginkgo/core/matrix/dense.hpp

```diff
@@ -154,6 +155,36 @@ class Dense
 public:
     using EnableLinOp<Dense>::convert_to;
     using EnableLinOp<Dense>::move_to;
+    using ConvertibleTo<Dense<next_precision<ValueType>>>::convert_to;
+    using ConvertibleTo<Dense<next_precision<ValueType>>>::move_to;
+    using ConvertibleTo<Coo<ValueType, int32>>::convert_to;
+    using ConvertibleTo<Coo<ValueType, int32>>::move_to;
+    using ConvertibleTo<Coo<ValueType, int64>>::convert_to;
+    using ConvertibleTo<Coo<ValueType, int64>>::move_to;
+    using ConvertibleTo<Csr<ValueType, int32>>::convert_to;
+    using ConvertibleTo<Csr<ValueType, int32>>::move_to;
+    using ConvertibleTo<Csr<ValueType, int64>>::convert_to;
+    using ConvertibleTo<Csr<ValueType, int64>>::move_to;
+    using ConvertibleTo<Ell<ValueType, int32>>::convert_to;
+    using ConvertibleTo<Ell<ValueType, int32>>::move_to;
+    using ConvertibleTo<Ell<ValueType, int64>>::convert_to;
+    using ConvertibleTo<Ell<ValueType, int64>>::move_to;
+    using ConvertibleTo<Fbcsr<ValueType, int32>>::convert_to;
+    using ConvertibleTo<Fbcsr<ValueType, int32>>::move_to;
+    using ConvertibleTo<Fbcsr<ValueType, int64>>::convert_to;
+    using ConvertibleTo<Fbcsr<ValueType, int64>>::move_to;
+    using ConvertibleTo<Hybrid<ValueType, int32>>::convert_to;
+    using ConvertibleTo<Hybrid<ValueType, int32>>::move_to;
+    using ConvertibleTo<Hybrid<ValueType, int64>>::convert_to;
+    using ConvertibleTo<Hybrid<ValueType, int64>>::move_to;
+    using ConvertibleTo<Sellp<ValueType, int32>>::convert_to;
+    using ConvertibleTo<Sellp<ValueType, int32>>::move_to;
+    using ConvertibleTo<Sellp<ValueType, int64>>::convert_to;
+    using ConvertibleTo<Sellp<ValueType, int64>>::move_to;
+    using ConvertibleTo<SparsityCsr<ValueType, int32>>::convert_to;
+    using ConvertibleTo<SparsityCsr<ValueType, int32>>::move_to;
+    using ConvertibleTo<SparsityCsr<ValueType, int64>>::convert_to;
+    using ConvertibleTo<SparsityCsr<ValueType, int64>>::move_to;
```

Comment:
could we mark `convert_to_impl` virtual and `convert_to` using the convert_to_impl in ConvertiableTo? (`convert_to` still need to be virtual to keep public interface)
then all class override `convert_to_impl` not `convert_to`

Reply:
This seems more complicated to me than the current solution. This is only temporary anyways, until we can change the virtual functions to take pointer_param directly with the next major release.